tools/schemapi/schemapi.py

```diff
@@ -1393,45 +1699,50 @@ def from_dict(
         """Construct an object from a dict representation."""
         target_tp: Any
         current_schema: dict[str, Any]
+        hash_schema: int
         if isinstance(dct, SchemaBase):
             return dct
         elif tp is not None:
             current_schema = tp._schema
+            hash_schema = _hash_schema(current_schema)
             root_schema: dict[str, Any] = rootschema or tp._rootschema or current_schema
```

Comment:
NOTE: the `current_schema` branch *only* is triggered for mock schema tests.

These all fail when `.. or current_schema` is removed:

`test_schemapi.py::[test_construct_multifaceted_schema, test_copy_method, test_round_trip, test_copy_module, test_from_dict, test_to_from_json, test_to_from_pickle]`